transmission/src/main/java/com/trendyol/transmission/transformer/Transformer.kt

```diff
@@ -87,10 +85,9 @@ open class Transformer(
                         .filter { it.identity == null || it.identity == internalIdentity }
                         .map { it.effect }
                         .collect {
-                            effectHandler?.apply {
-                                currentEffectProcessing = launch {
-                                    communicationScope.onEffect(it)
-                                }
+                            currentEffectProcessing = launch {
```

Comment:
Why don't we do the same scoping for effect processing 🤔

Reply:
We can use either `transformerScope.launch` or wrap coroutine builders inside a `supervisorScope`.  I preferred the `supervisorScope` cause there are multiple launch statements; instead of adding a `transformerScope` prefix to all of the builders, just wrapping up would look clearer, IMO.